src/core/qgsvectorfilewriter.cpp

```diff
@@ -2831,7 +2831,7 @@ bool QgsVectorFileWriter::deleteShapeFile( const QString &fileName )
   QDir dir = fi.dir();
 
   QStringList filter;
-  const char *suffixes[] = { ".shp", ".shx", ".dbf", ".prj", ".qix", ".qpj" };
+  const char *suffixes[] = { ".shp", ".shx", ".dbf", ".prj", ".qix", ".qpj", ".cpg", ".sbn", ".sbx", ".idm", ".ind" };
```

Comment:
@agiudiceandrea would you be ok with taking the possibility to update this to a more Qt-style

```cpp
  static const QStringList suffixes = { QStringLiteral( ".shp" ), ... };
  for ( const QString &suffix : suffixes )
  {
    ...
  }
```